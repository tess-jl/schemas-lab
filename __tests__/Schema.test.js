const { Schema } = require('../lib/Schema.js');

const schema = new Schema({
  name: {
    type: String,
    required: true
  }, 
  age: {
    type: Number,
    required: true
  },
  weight: {
    type: String
  }
});
  
const spot = {
  name: 'spot',
  age: 5,
  weight: '20 lbs'
};
  
const rover = {
  name: 'rover',
  age: '10'
};
  
const who = {
  age: 'hi'
};


describe('Schema class tests', () => {
  describe('Schema class method tests', () => {
    it('returns the object with all fields cast', () => {
      expect(schema.validate(spot)).toEqual({
        name: 'spot',
        age: 5,
        weight: '20 lbs'
      });
      expect(schema.validate(rover)).toEqual({
        name: 'rover',
        age: 10
      });
    });
    it('throws an error if the object doesn\'t follow the schema', () => {
      expect(schema.validate(who)).toThrowErrorMatchingSnapshot();
    });
  });
});
