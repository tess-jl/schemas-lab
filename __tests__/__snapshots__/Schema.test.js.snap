// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema class tests Schema class method tests throws an error if the object doesn't follow the schema 1`] = `"received is not a function"`;
